aesm-client/src/win_aesm_client.rs

```diff
@@ -0,0 +1,349 @@
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub proxy_setting_assist: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            proxy_info: *mut u8,
+            proxy_size: u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub query_sgx_status: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            sgx_status: *mut u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub get_whitelist_size: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            white_list_size: *mut u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub get_white_list: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            white_list: *mut u8,
+            buf_size: u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub get_sec_domain_id: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            sec_domain_id: *mut u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub switch_sec_domain: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            sec_domain_id: u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub get_epid_provision_status: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            epid_pr_status: *mut u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+    pub get_platform_service_status: ::std::option::Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterfaceT,
+            pse_status: *mut u32,
+            result: *mut AesmErrorT,
+        ) -> HRESULT,
+    >,
+}
+
+#[cfg(windows)] impl Drop for AesmClient {
+    fn drop (&mut self ) {
+        unsafe {
+            if let Some(release) = (*(*self.interface).vtbl).release {
+                release(self.interface);
+            }
+            CoUninitialize();
+        }
+    }
+}
+
+#[cfg(windows)]
+impl AesmClient {
+    pub fn new () -> Result<Self> {
+        let aesm_client = AesmClient::create_instance()?;
+        Ok(aesm_client)
+    }
+
+    fn create_instance() -> Result<Self> {
+        let mut instance : *mut AesmInterfaceT = std::ptr::null_mut();
+        let clsid_aesminterface : CLSID = CLSID {
+            Data1: 0x82367CAB,
+            Data2: 0xF2B9,
+            Data3: 0x461A,
+            Data4: [0xB6, 0xC6, 0x88, 0x9D, 0x13, 0xEF, 0xC6, 0xCA]
+        };
+        let iid_iaesminterface : IID = IID {
+            Data1: 0x50AFD900,
+            Data2: 0xF309,
+            Data3: 0x4557,
+            Data4: [0x8F, 0xCB, 0x10, 0xCF, 0xAB, 0x80, 0x2C, 0xDD]
+        };
```

Comment:
Make these two definitions `const` and put them somewhere near the vtbl definition.